src/transformers/utils/fx.py

```diff
@@ -22,16 +22,88 @@
     MODEL_FOR_SEQ_TO_SEQ_CAUSAL_LM_MAPPING,
     MODEL_FOR_SEQUENCE_CLASSIFICATION_MAPPING,
     MODEL_FOR_TOKEN_CLASSIFICATION_MAPPING,
+    MODEL_MAPPING,
     GPT2DoubleHeadsModel,
+    PretrainedConfig,
     PreTrainedModel,
     logging,
 )
+from ..file_utils import TORCH_FX_REQUIRED_VERSION, importlib_metadata, is_torch_fx_available
 from ..models.auto import get_values
+from .fx_transformations import (
+    _cache_attributes,
+    _patch_arguments_,
+    _restore_attributes_,
+    transform_to_dynamic_input_,
+    transformation,
+)
 
 
 logger = logging.get_logger(__name__)
 
 
+T = TypeVar("T")
+TypeOrListOfType = Union[T, List[T]]
+
+
+def _generate_supported_model_classes(
+    model_name: Type[PretrainedConfig],
+    supported_tasks: Optional[TypeOrListOfType[str]] = None,
+):
+    model_config_class = CONFIG_MAPPING[model_name]
+    task_mapping = {
+        "default": MODEL_MAPPING,
+        "pretraining": MODEL_FOR_PRETRAINING_MAPPING,
+        "next-sentence-prediction": MODEL_FOR_NEXT_SENTENCE_PREDICTION_MAPPING,
+        "masked-lm": MODEL_FOR_MASKED_LM_MAPPING,
+        "causal-lm": MODEL_FOR_CAUSAL_LM_MAPPING,
+        "seq2seq-lm": MODEL_FOR_SEQ_TO_SEQ_CAUSAL_LM_MAPPING,
+        "multiple-choice": MODEL_FOR_MULTIPLE_CHOICE_MAPPING,
+        "question-answering": MODEL_FOR_QUESTION_ANSWERING_MAPPING,
+        "sequence-classification": MODEL_FOR_SEQUENCE_CLASSIFICATION_MAPPING,
+        "token-classification": MODEL_FOR_TOKEN_CLASSIFICATION_MAPPING,
+        "image-classification": MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING,
+    }
+
+    if supported_tasks is None:
+        supported_tasks = task_mapping.keys()
+    if isinstance(supported_tasks, str):
+        supported_tasks = [supported_tasks]
+
+    model_classes = []
+    for task in supported_tasks:
+        model_class = task_mapping[task].get(model_config_class, None)
+        if model_class:
+            model_classes.append(model_class)
+
+    return tuple(model_classes)
+
+
+_REGULAR_SUPPORTED_MODELS = (
+    *_generate_supported_model_classes("albert"),
+    *_generate_supported_model_classes("bert"),
+    *_generate_supported_model_classes("distilbert"),
+    *_generate_supported_model_classes("mobilebert"),
+    *_generate_supported_model_classes("electra"),
+    *_generate_supported_model_classes("megatron-bert"),
+    *_generate_supported_model_classes("gpt2"),
+    *_generate_supported_model_classes("gptj"),
+    *_generate_supported_model_classes("gpt_neo"),
+    *_generate_supported_model_classes("t5"),
```

Comment:
I would still make a list with those supported model types and then build `_REGULAR_SUPPORTED_MODELS` form it, just so we can make it easier to extend the support ;-)